### PR TITLE
Docbash

### PR DESCRIPTION
**Pull request by zackchase**
Architecture docs (index.md and program_model.md) rewrite.

### CODE REVIEW

File: docs/how_to/multi_devices.md [hunk 1]
```diff
@@ -23,7 +23,11 @@ For the duration of this tutorial, we'll focus on data parallelism.
 
 ### Workload Partitioning
 
+<<<<<<< HEAD
```
Comment (sergeykolychev) [inline docs/how_to/multi_devices.md]: artifacts from some merge.

File: docs/how_to/index.md [hunk 2]
```diff
@@ -37,3 +37,13 @@ and full working examples, visit the [tutorials section](../tutorials/index.md).
...
 
 * [How do I run MXNet on a Raspberry Pi for computer vision?](http://mxnet.io/tutorials/embedded/wine_detector.html)
+
+## Extend and Contribute to MXNet
+
+* [How do I contribute a patch to MXNet?](http://mxnet.io/community/contribute.html)
+
+* [How do I create new operators in MXNet?](http://mxnet.io/how_to/new_op.html)
+
+* [How do I set MXNet's environmental variables?](http://mxnet.io/how_to/env_var.html)
```
Comment (piiswrong) [inline docs/how_to/index.md]: Can we make it more explicit about what environmental variables are for?

Comment (zackchase) [inline docs/how_to/index.md]: In the title itself?

Comment (piiswrong) [inline docs/how_to/index.md]: yeah. Its mostly related to performance. We should make it more explicit in the title.

Comment (piiswrong) [inline docs/how_to/index.md]: For example
How do I set MXNet's environmental variables for better performance?

Comment (zackchase) [inline docs/how_to/index.md]: That wouldn't be accurate. The document doesn't really address performance. If we want to give the document a sexier title, then we need to write a more useful document. Really, it's just a reference. We can deal with this in the future. For now, please merge.

**Pull request merged**